src/auto-evo/simulation/food_source/ChunkFoodSource.cs

```diff
@@ -38,6 +46,13 @@ public override float FitnessScore(Species species, SimulationCache simulationCa
 
         score /= energyBalance.FinalBalanceStationary;
 
+        // We ponder the score for each compound by its amount, leading to pondering in proportion of total quantity,
+        // with a constant factor that will be eliminated when making ratios of scores for this niche.
+        var ponderedEnergyGenerationScore = energyCompounds.Sum(
```

Comment:
I think this doesn't need a separate variable. Also this is part of the *base* score, so to me this multiplying is pretty weird here. Instead I'd move this to line 39 and make this an addition.

Reply:
It is not part of the base score, it *is* the base score. All the rest are multiplicative factors to this base.

Reply:
Yes, but you *multiply* it into the score *after* all the other stuff has been already calculated, at least github diff shows it to me like that.

Reply:
![Kuvakaappaus - 2021-12-09 14-08-29](https://user-images.githubusercontent.com/3796411/145393823-9b03a4a9-9559-4957-972c-0c62df972561.png)

Reply:
I've just been reordering this, because I agree this may be misleading. But I *think* I saw it in another food source. Not sure though.

Reply:
Still you are *multiplying* the *final* score after all the adjustments, with what should be the *base* number and the adjustments should apply on top of that number.

Reply:
As I wrote just above, I already reordered that part of the logic.